File: api/trades/stock/[symbol].js
function generateMockData(symbol) {
  const data = [];
  let basePrice = 50 + Math.random() * 450;
  const now = new Date();

  for (let i = 30; i >= 0; i--) {
    const date = new Date(now);
    date.setDate(date.getDate() - i);

    const change = (Math.random() - 0.5) * 10;
    basePrice += change;
    const open = basePrice;
    const close = basePrice + (Math.random() - 0.5) * 5;
    const high = Math.max(open, close) + Math.random() * 3;
    const low = Math.min(open, close) - Math.random() * 3;

    data.push({
      time: date.toISOString().split('T')[0],
      open: Math.round(open * 100) / 100,
      high: Math.round(high * 100) / 100,
      low: Math.round(low * 100) / 100,
      close: Math.round(close * 100) / 100,
      volume: Math.floor(Math.random() * 10000000) + 1000000,
    });
  }

  return data;
}

export default function handler(req, res) {
  res.setHeader('Access-Control-Allow-Origin', '*');

  if (req.method === 'OPTIONS') {
    return res.status(200).end();
  }

  const { symbol } = req.query;
  const data = generateMockData(symbol);

  res.json(data);
}

File: api/trades/stock/[symbol].py
from http.server import BaseHTTPRequestHandler
import json
import urllib.parse
from datetime import datetime, timedelta
import random


def generate_mock_data(symbol):
    """Generate mock stock data"""
    data = []
    base_price = random.uniform(50, 500)

    for i in range(30):
        date = (datetime.now() - timedelta(days=30-i)).strftime("%Y-%m-%d")
        daily_change = random.uniform(-0.03, 0.03)
        open_price = base_price * (1 + daily_change)
        high = open_price * (1 + random.uniform(0, 0.02))
        low = open_price * (1 - random.uniform(0, 0.02))
        close = random.uniform(low, high)

        data.append({
            "time": date,
            "open": round(open_price, 2),
            "high": round(high, 2),
            "low": round(low, 2),
            "close": round(close, 2),
            "volume": random.randint(1000000, 50000000)
        })

        base_price = close

    return data


class handler(BaseHTTPRequestHandler):
    def do_GET(self):
        # Extract symbol from path
        path_parts = self.path.split('/')
        symbol = urllib.parse.unquote(path_parts[-1].split('?')[0]).upper()

        # Generate mock data
        data = generate_mock_data(symbol)

        self.send_response(200)
        self.send_header('Content-type', 'application/json')
        self.send_header('Access-Control-Allow-Origin', '*')
        self.end_headers()
        self.wfile.write(json.dumps(data).encode())

    def do_OPTIONS(self):
        self.send_response(200)
        self.send_header('Access-Control-Allow-Origin', '*')
        self.send_header('Access-Control-Allow-Methods', 'GET, OPTIONS')
        self.send_header('Access-Control-Allow-Headers', 'Content-Type')
        self.end_headers()
